ruma-api-macros/src/api.rs

```diff
@@ -83,8 +83,19 @@ pub fn expand_all(api: Api) -> syn::Result<TokenStream> {
     // with only the literal's value from here on.
     let name = &api.metadata.name.value();
     let path = &api.metadata.path;
-    let rate_limited = &api.metadata.rate_limited;
-    let authentication = &api.metadata.authentication;
+    let rate_limited = &api
+        .metadata
+        .rate_limited
+        .iter()
+        .map(|r| {
+            let attrs = &r.attrs;
+            let value = &r.value;
+            quote! {
+                #( #attrs )*
+                rate_limited: #value,
+            }
+        })
+        .collect::<Vec<_>>();
```

Comment:
You can collect this into a `TokenStream` directly (turbofish not needed if you annotate the binding), then you don't need the `#()` below.

```suggestion
        .collect();
```